.github/workflows/validation-nightly.yaml

```diff
@@ -0,0 +1,57 @@
+name: Snapshots Validation
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 1 * * *'
+jobs:
+  snapshot-builds:
+    strategy:
+      fail-fast: false
+      matrix:
+        hilla: ["2.2-SNAPSHOT", "2.1-SNAPSHOT", "1.3-SNAPSHOT"]
+        java: [ 17, 20 ]
+        branch: [ "main", "1.0" ]
+        include:
+          - hilla: "1.3-SNAPSHOT"
+            java: 11
+            branch: "1.0"
+        exclude:
+          - hilla: "1.3-SNAPSHOT"
+            branch: "main"
+          - hilla: "2.1-SNAPSHOT"
+            branch: "1.0"
+          - hilla: "2.2-SNAPSHOT"
+            branch: "1.0"
```

Comment:
I would split this into two jobs, one for 1.0 and one for main, this looks a bit hacky.

Reply:
Do you mean different workflows? Actually the matrix creates a job for every combination

Reply:
Just anything to not have this include/exclude stuff, so that it may be shorter. But if this is already the best solution for that, I am fine.

Reply:
Now I got it. We can have two jobs, one per branch, and both will have a hilla/java matrix.
I'll try it in the next days

Reply:
Ok Great. Maybe the rest can be a template. We can also merge this now and create a follow up for improving this?

Reply:
Sounds good. Let's merge as is, so we can have early feedback on potential failures with Hilla updates. Then we can improve the workflow